backend/convert_from_protobuf_test.go

```diff
@@ -533,3 +533,83 @@ func datasourceInstanceProtoFieldCountDelta() int64 {
 	// returning 1 to account for the Type field in the SDK that is not in the protobuf
 	return int64(1)
 }
+
+func TestConvertFromProtobufAdmissionRequest(t *testing.T) {
+	protoPIS := &pluginv2.AdmissionRequest{
+		PluginContext: protoPluginContext,
+		Operation:     pluginv2.AdmissionRequest_CREATE,
+		Kind: &pluginv2.GroupVersionKind{
+			Group:   "g",
+			Version: "v",
+			Kind:    "k",
+		},
+		ObjectBytes: []byte(`{}`),
+	}
+
+	protoWalker := &walker{}
+	err := reflectwalk.Walk(protoPIS, protoWalker)
+	require.NoError(t, err)
+
+	// if protoWalker.HasZeroFields() {
+	// 	t.Fatalf(unsetErrFmt,
+	// 		"proto", "InstanceSettingsRequest", protoWalker.ZeroValueFieldCount, protoWalker.FieldCount)
+	// }
+
+	// TODO??
```

Comment:
I guess this still need some work

Reply:
ya --- reluctant to do so much handcrafted mess while still finding the right API structure 😛

Reply:
then you can delete it for now?

Reply:
I'll add the tests before merge -- they just require hand checking each property...

Now that I think we are close to the final structure -- i'll add them